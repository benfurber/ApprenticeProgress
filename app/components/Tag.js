import React, { Component } from "react";
import { Badge, Text } from "native-base";

import { lastWordOfString } from "utils";

class Tag extends Component {
  render() {
    const { text } = this.props;

    return (
      <Badge style={{ backgroundColor: this._tagStyle(text) }}>
        <Text>{text}</Text>
      </Badge>
    );
  }

  _tagStyle(text) {
    const option = lastWordOfString(text);
    const styleOptions = {
      thoughtbot: "#ED3E44",
      developer: "#EF8D3F",
      consultant: "#BE3284",
    };

    return styleOptions[option];
  }
}

export { Tag };
