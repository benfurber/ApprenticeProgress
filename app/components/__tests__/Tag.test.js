// @flow

import React from "react";
import { shallow } from "enzyme";

import { Tag } from "../Tag";

describe("Tag", () => {
  it("has a valid snapshot", () => {
    const component = shallow(<Tag text="I'm team thoughtbot" />);

    expect(component).toMatchSnapshot();
  });
});
