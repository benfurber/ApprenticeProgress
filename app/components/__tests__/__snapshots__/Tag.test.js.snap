// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    text="I'm team thoughtbot"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Text)>
        I'm team thoughtbot
      </Styled(Text)>,
      "style": Object {
        "backgroundColor": "#ED3E44",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "I'm team thoughtbot",
        "virtual": undefined,
      },
      "ref": null,
      "rendered": "I'm team thoughtbot",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Text)>
          I'm team thoughtbot
        </Styled(Text)>,
        "style": Object {
          "backgroundColor": "#ED3E44",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "I'm team thoughtbot",
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "I'm team thoughtbot",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
